workspaces/mcp-integrations/plugins/backstage-mcp-plugin-backend/src/plugin.ts

```diff
@@ -41,33 +41,123 @@ export const backstageMcpPlugin = createBackendPlugin({
       },
       // Sample action used in the Backstage docs: https://github.com/backstage/backstage/tree/master/plugins/mcp-actions-backend
       async init({ actionsRegistry, catalog, auth }) {
-        actionsRegistry.register({
-          name: 'greet-user',
-          title: 'Greet User',
-          description: 'Generate a personalized greeting',
-          schema: {
-            input: z =>
-              z.object({
-                name: z.string().describe('The name of the person to greet'),
-              }),
-            output: z =>
-              z.object({
-                greeting: z.string().describe('The generated greeting'),
-              }),
-          },
-          action: async ({ input }) => ({
-            output: { greeting: `Hello ${input.name}!` },
-          }),
-        });
         // This action is used to fetch the list of catalog entities from Backstage. It returns an array of entity names
         actionsRegistry.register({
           name: 'fetch-catalog-entities',
           title: 'Fetch Catalog Entities',
-          description:
-            'Retrieve the list of catalog entities from the Backstage server.',
+          description: `Search and retrieve catalog entities from the Backstage server.
+
+List all Backstage entities such as Components, Systems, Resources, APIs, Locations, Users, and Groups. 
+Results are returned in JSON array format, where each entry in the JSON array has the following fields: 'name', 'description','uid', and 'type'.
+
+This tool searches through the software catalog to find components the entities. It supports filtering by entity properties and 
```

Comment:
`to find components the entities` does not parse as grammatically correct to me @johnmcollier but I'm not sure where you were going, so I couldn't just suggest the correction

Reply:
also down below you got some more precise restrictions on what you can filter on ... i.e. the comment below that starts with `Validate that type is only used with kind -- we could just allow 'type' to be specified without 'kind' but given types...`